File: code/c/storing-using-data/test-fn-calls.c
/* Program: test-fn-calls.c */
#include <stdio.h>

int square(int x) { return x * x; }
int sum(int a, int b) { return a + b; }

int main()
{
    int answer = sum(square(5), square(4));
    printf("5 squared + 4 squared is %d\n", answer);
    
    printf("(1 + 2) + (3 + 4) = %d\n", sum(sum(1, 2), sum(3, 4)));
    printf("2 squared, squared = %d\n", square(square(2)));
    
    return 0;
}

File: code/pascal/storing-using-data/TestFnCalls.pas
program TestFnCalls;

function Square(x: Integer): Integer;
begin
    result := x * x;
end; 

function sum(a, b: Integer): Integer;
begin
    result := a + b;
end;

procedure Main();
var
    answer: Integer;
begin
    answer := sum(square(5), square(4));
    WriteLn('5 squared + 4 squared is ', answer);
    
    WriteLn('(1 + 2) + (3 + 4) = ', Sum(Sum(1, 2), Sum(3, 4)));
    WriteLn('2 squared, squared = ', Square(Square(2)));
end;

begin
    Main();
end.